Dockerfile

```diff
@@ -1,10 +1,11 @@
-FROM registry.suse.com/bci/golang:1.23
+FROM registry.suse.com/bci/golang:1.23 AS builder_base
```

Comment:
The `builder_base` stage is the former `Dockerfile.dapper`, and the rest stages are the content of `package/Dockerfile` originally.

- `builder_base` stage is the target stage of dapper for local build
- `builder_base` stage is also the base of `engine_builder` for self-contained building
- Due to dapper issue, we need to move Dockerfile from package to repo root: [FATA[0000] pattern contains path separator](https://github.com/rancher/dapper/issues/92)